src/lib/integration_test_cloud_engine/mina_automation.ml

```diff
@@ -879,7 +884,20 @@ module Network_manager = struct
     let%bind () = File_system.remove_dir t.testnet_dir in
     Deferred.unit
 
-  let destroy t =
+  let tear_down t network =
+    if t.generate_code_coverage then
+      let open Malleable_error.Let_syntax in
+      let coverage_manager = Coverage_manager.create ~logger:t.logger in
+      let%bind _ =
+        Coverage_manager.download_coverage_data_from_nodes coverage_manager
+          ~nodes:(Kubernetes_network.all_nodes network)
+      in
+      Malleable_error.ok_unit
+    else Malleable_error.ok_unit
+
+  let destroy t network =
```

Comment:
Revert, call the `generate_code_coverage` helper from the `deploy` definition above instead.